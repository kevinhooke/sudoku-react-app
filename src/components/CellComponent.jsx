import React, { Component } from 'react';

class CellComponent extends Component {

    constructor(props) {
        super(props);

        this.handleChange = this.handleChange.bind(this);
    }

    handleChange(event){
        this.props.onChange(event);
    }

    render() {
        let cell = null;
        if(this.props.value.initialGiven){
            cell = <div>{this.props.value.value}</div>
        }
        else{
            cell = <input className="cell" type="text" value={this.props.value.value} onChange={this.handleChange}/>
        }

        return (
            <div className="label">
                { cell }
            </div>
        );
    }

}
export default CellComponent;