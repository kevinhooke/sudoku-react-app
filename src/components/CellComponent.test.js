import React from 'react';
import renderer, { create }  from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import CellComponent from './CellComponent.jsx';

describe('Renders a Cell', () => {

    it('Renders cell with initial given value of 1 - snapshot compare', () => {
        let value = {
            value: "1"
        }
        const component = renderer.create(<CellComponent value={value}/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('Renders input, initialGiven = false', () => {
        let value = {
            value: "1",
            initialGiven: false
        }
        const renderer = new ShallowRenderer();
        renderer.render(<CellComponent value={value}/>);
        const result = renderer.getRenderOutput();

        //console.log(result);
        expect(result.type).toBe('div');

        // expected: <input className="cell" onChange={handleChange} type="text" value="1" />
        expect(result.props.children.type).toBe("input");
        expect(result.props.children.props.className).toBe("cell");
        expect(result.props.children.props.value).toBe("1");
    })

    it('Renders div, initialGiven = true', () => {
        let value = {
            value: "1",
            initialGiven: true
        }
        const renderer = new ShallowRenderer();
        renderer.render(<CellComponent value={value}/>);
        const result = renderer.getRenderOutput();

        //console.log(result);
        expect(result.type).toBe('div');
        expect(result.props.children.type).toBe("div");
        expect(result.props.children).toEqual(
            <div>1</div>
        );
    })
})