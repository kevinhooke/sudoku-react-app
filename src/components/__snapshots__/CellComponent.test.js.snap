// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a Cell Renders cell with initial given value of 1 - snapshot compare 1`] = `
<div
  className="label"
>
  <input
    className="cell"
    onChange={[Function]}
    type="text"
    value="1"
  />
</div>
`;
